### ISSUE TITLE
Error fixes are logged without indicating file names

### ISSUE THREAD

**Issue by devend711** [charliermarsh/ruff#1667]
Example output from running `ruff` via a git pre-commit hook:

```
ruff.....................................................................Failed
- hook id: ruff
- files were modified by this hook

Found 1 error(s) (1 fixed, 0 remaining).
```

Locally, it's possible to do something like `git status` to see which files were changed as a result of the hook. But when these hooks are run in a CI run for example, seeing the file names in ruff's output would be useful

**Comment by charliermarsh** [comment]
Is there any way to enable this in pre-commit?

**Comment by charliermarsh** [comment]
Oh, I guess this has nothing to do with pre-commit, since we don't actually log the names of fixed files. I wonder if we should.

**Comment by telamonian** [comment]
@charliermarsh I'm still having this same problem. When I run eg `git commit -a` I see:

```
ruff.....................................................................Failed
- hook id: ruff
- files were modified by this hook

Found 1 error (1 fixed, 0 remaining).

ruff-format..............................................................Failed
- hook id: ruff-format
- files were modified by this hook

1 file reformatted, 1 file left unchanged
```

Especially when you try and commit with `git commit -a`, there's currently no way to see what files changed or what the changes actually were. Am I missing something, like how to apply the existing fix you linked above?

**Comment by dhruvmanila** [comment]
@telamonian Can you try using the `--show-fixes` flag with the pre-commit hook? It'll show you the files (and rules) that were fixed:
```
Fixed 4 errors:
- src/lsp.py:
    3 × F401 (unused-import)
    1 × COM812 (missing-trailing-comma)

Found 8 errors (4 fixed, 4 remaining).
```

So, I think the following should give you what you want:
```yaml
- repo: https://github.com/astral-sh/ruff-pre-commit
  rev: v0.6.3
  hooks:
    - id: ruff
      args: [ --show-fixes ]
```

**Comment by telamonian** [comment]
@dhruvmanila Hey, sorry I never noticed that you actually responded to my original post. Your suggestion of using `--show-fixes` solves part of my problem, but not all of it. I do now see the fixes caused by the `ruff` hook, but then any fixes made by the `ruff-format` hook are still hidden:

```
$ git commit -a
ruff.....................................................................Passed
ruff-format..............................................................Failed
- hook id: ruff-format
- files were modified by this hook

2 files reformatted, 1 file left unchanged
```

Is there an equivalent of `show-fixes` for the `ruff format` command? I did already try just passing `show-fixes`, but it is not recognized.

**Comment by dhruvmanila** [comment]
> Is there an equivalent of `show-fixes` for the `ruff format` command? I did already try just passing `show-fixes`, but it is not recognized.

There's no equivalent to "format AND show diff" on the formatter side but you can do either of those i.e., use `--diff` to get the diff between original and formatter source but that wouldn't change the file content on disk. https://github.com/astral-sh/ruff/issues/8191 seems related.
